### PR TITLE
Added distribution lookup option --base-path

### PR DESCRIPTION
**Pull request by gerrod3**
fixes: #134

### CODE REVIEW

File: CHANGES/134.feature [hunk 1]
```diff
@@ -0,0 +1,2 @@
+Distributions can now be found with option --base-path for list and show commands
+
```
Comment (daviddavis) [inline CHANGES/134.feature:2]:
```suggestion
```

File: tests/scripts/pulp_file/test_distribution.sh [hunk 2]
```diff
@@ -37,6 +38,12 @@ expect_succ pulp file distribution update \
   --base-path "cli_test_file_distro" \
   --publication "$PUBLICATION_HREF"
 
+
+expect_succ pulp file distribution show --base-path "cli_test_file_distro"
+test "$(echo "$OUTPUT" | jq -r .pulp_href)" = "$DISTRIBUTION_HREF"
+expect_succ pulp file distribution list --base-path "cli"
+test "$(echo "$OUTPUT" | jq -r .[0].pulp_href)" = "$DISTRIBUTION_HREF"
```
Comment (daviddavis) [inline tests/scripts/pulp_file/test_distribution.sh:45]: This seems very brittle--I often run this test against my dev server which may have other file distributions. Maybe we could just check that the number of results is > 0?

**Comment by daviddavis** [comment]
I had imagined that the `--base-path` would just be a filter on list. But now that I think about it, I can't see any downside I suppose to having it on show.

File: tests/scripts/pulp_file/test_distribution.sh (hunk 2 above)
Comment (mdellweg) [reply tests/scripts/pulp_file/test_distribution.sh:45]: If the filter `--base-path "cli"` is working correctly, there must only be a single item returned. So we could even check for cout==1, no?

File: pulpcore/cli/common/generic.py [hunk 3]
```diff
@@ -112,6 +112,19 @@ def load_json_callback(
...
+def _base_path_callback(
+    ctx: click.Context, param: click.Parameter, value: Optional[str]
+) -> Optional[str]:
+    if value is not None:
+        if ctx.info_name == "show":
+            entity_ctx: PulpEntityContext = ctx.find_object(PulpEntityContext)
+            entity_ctx.entity = {"base_path": value}
+        elif ctx.info_name == "list":
+            param.name = "base_path__icontains"
+            param.expose_value = True
```
Comment (mdellweg) [inline pulpcore/cli/common/generic.py:124]: I'd say, if you want a base_path__icontains filter, that make it a --base-path-contains option.

File: tests/scripts/pulp_file/test_distribution.sh (hunk 2 above)
Comment (daviddavis) [reply tests/scripts/pulp_file/test_distribution.sh:45]: That's true altough the filter is `base_path__icontains` though so there's still a chance.

**Comment by daviddavis** [comment]
We talked about this PR at our pulp-cli meeting and we thought maybe the `--base-path` option on show didn't make sense since we're not going to offer it on other endpoints such as update. It seems more of a filter than a lookup option?

**Review by daviddavis — APPROVED** [review]

**Review by mdellweg — APPROVED** [review]

**Pull request merged**